### PR TITLE
Make term nullability explicit

### PR DESCRIPTION
**Pull request by ralfhandl**

### CODE REVIEW

**Review by BerSie — DISMISSED** [review]
From my point of view OK - I mainly checked Common, Communication, PersonalData and UI. But what's the reason for doing it?

**Comment by ralfhandl** [comment]
@qmacro stumbled across `HeaderInfo` being nullable, which doesn't make much sense to me, and is confusing for readers of the vocabulary.

File: vocabularies/UI.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@ a collection of business object instances, e.g. as a list or table.</String>
 
       <!-- Semantic Views / Perspectives -->
 
-      <Term Name="HeaderInfo" Type="UI.HeaderInfoType" AppliesTo="EntityType">
+      <Term Name="HeaderInfo" Type="UI.HeaderInfoType" Nullable="false" AppliesTo="EntityType">
```
Comment (HeikoTheissen) [inline vocabularies/UI.xml:58]: If I want an Employee's header info to be the same as their Manager's header info, I could write:
```jsonc
"Employee": {
  "$Kind": "EntityType",
  "$BaseType": "self.Manager",
  "Manager": {
    "$Kind": "NavigationProperty",
    "$Type": "self.Manager"
  },
  "@UI.HeaderInfo": {"$Path": "Manager/@UI.HeaderInfo"}, ...
}
```
But if the Employee has no Manager, the `$Path` would evaluate to a forbidden null value.

Shouldn't annotations that might be used with dynamic values be nullable?

Comment (ralfhandl) [reply vocabularies/UI.xml:58]: Correct, but does that make sense for HeaderInfo? How would an Object Page without HeaderInfo look like?

Comment (ralfhandl) [reply vocabularies/UI.xml:58]: Terms that legally can have a null value, for example Common.FilterDefaultValue, are now explicitly marked as Nullable.

Same for terms that are explicitly defined to be used with path expressions.

Comment (HeikoTheissen) [reply vocabularies/UI.xml:58]: > Correct, but does that make sense for HeaderInfo? How would an Object Page with a null HeaderInfo look like?

The client could use a default, for example list all fields alphabetically.

Comment (ralfhandl) [reply vocabularies/UI.xml:58]: @BerSie what will the client "Fiori Elements" do if it encounters `@UI.HeaderInfo: null`?

**Review by qmacro — COMMENTED** [review]
From a consumer / reader POV this change makes a lot of sense to me, thanks. I'd wondered about the `?` suffix on types for the terms in our vocabs and remained vaguely confused for a good while.

**Comment by qmacro** [comment]
Background on where my questions about the `?` suffix came from - writing a small guide to understanding annotations, specifically this bit https://github.com/SAP-samples/odata-basics-handsonsapdev/blob/annotations/bookshop/README.md#the-ui-annotation-vocabulary

**Review by qmacro — DISMISSED** [review]
LGTM, thanks!

File: vocabularies/UI.xml (hunk 1 above)
Comment (BerSie) [reply vocabularies/UI.xml:58]: There might be places where this is handled gracefully, but I wouldn't bet on that. Already today the vocabulary states that the HeaderInfo is mandatory for the "main entity types". And it's not only properties , where can take it e.g. alphabetically from a list, it's also the object type name. From a consumption (and UX) perspective I regards the header information as mandatory.

**Review by GeraldKrause — COMMENTED** [review]
Checked Analytics vocabulary; looks good to me

**Comment by ralfhandl** [comment]
Merged `main` branch because `Common.DefaultValuesFunction` is no longer experimental, see #189

**Review by qmacro — APPROVED** [review]

**Review by HeikoTheissen — APPROVED** [review]

**Pull request merged**